diff --git a/pkg/banking/banking.go b/pkg/banking/banking.go
index 0c235c8..3257d9b 100644
--- a/pkg/banking/banking.go
+++ b/pkg/banking/banking.go
@@ -175,7 +175,7 @@ func (c *Client) GetBalance(accessToken string, bankID string, ctx context.Conte
 }
 
 func (c *Client) GetBankStatus(bankID string) (*plaid.Institution, error) {
-	var ctx context.Context
+	ctx := context.Background()
 	bankConfig := c.Banks[bankID]
 	req := plaid.InstitutionsGetByIdRequest{
 		InstitutionId: bankConfig.Institution,
