diff --git a/api/services/sheets_service/budget_service.go b/api/services/sheets_service/budget_service.go
index 2dfcfbb..3be8a1c 100644
--- a/api/services/sheets_service/budget_service.go
+++ b/api/services/sheets_service/budget_service.go
@@ -72,7 +72,7 @@ func (ss *SheetsService) ReadBudgetSheet() (*BudgetSheet, error) {
 }
 
 func (ss *SheetsService) isEmptyBudgetRow(values []interface{}) bool {
-	if len(values) < 6 || ss.getBudgetCategory(values) == "" {
+	if len(values) < 7 || ss.getBudgetCategory(values) == "" {
 		return true
 	}
 	return false
